### PR TITLE
Clean up formatting of Forge Test mods and disable noisy ones

### PR DESCRIPTION
**Pull request by mezz**
Cleaning up the test mods sets a better example when people go to make new ones.
The log spam was also getting pretty intense, making it hard to check new test mods.

The difference between `net.minecraftforge.test` and `net.minecraftforge.debug` was ambiguous, with people putting things in at random, so I moved all `@Mod` tests to `debug` and left the JUnit tests in `test`.

### CODE REVIEW

File: src/test/java/net/minecraftforge/debug/BlockAiNodeTypeTest.java [hunk 1]
```diff
@@ -10,30 +8,33 @@
...
+    {
 
-        TestBlock() {
+        TestBlock()
+        {
             super(Material.ROCK);
             setRegistryName("testblock");
         }
 
-        @Nullable
```
Comment (liach) [inline src/test/java/net/minecraftforge/debug/BlockAiNodeTypeTest.java:36]: Why?

File: src/test/java/net/minecraftforge/debug/EnumPlantTypeTest.java [hunk 2]
```diff
@@ -27,11 +33,15 @@ public void onInit(FMLInitializationEvent event)
...
             if (biomeType == null || !biomeType.name().equals("FAKE"))
+            {
                 LOGGER.warn("EnumHelper in BiomeType is working incorrectly!");
+            }
         }
         EnumPlantType plantType = null;
         if (plantType == null || !plantType.name().equals("FAKE"))
+        {
             ;
+        }
```
Comment (liach) [inline src/test/java/net/minecraftforge/debug/EnumPlantTypeTest.java]: Why do you even keep this block when it does not have an `else` statement after?

File: src/test/java/net/minecraftforge/debug/ItemTileDebug.java [hunk 3]
```diff
@@ -81,7 +93,11 @@ public void init(FMLInitializationEvent event) {
     {
         public static final BakeEventHandler instance = new BakeEventHandler();
 
-        private BakeEventHandler() {};
+        private BakeEventHandler()
+        {
+        }
+
+        ;
```
Comment (liach) [inline src/test/java/net/minecraftforge/debug/ItemTileDebug.java]: Looks useless.

File: src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java [hunk 4]
```diff
@@ -48,24 +39,35 @@
...
+import java.util.ArrayList;
+import java.util.List;
 
 @Mod(modid = ModelLoaderRegistryDebug.MODID, name = "ForgeDebugModelLoaderRegistry", version = ModelLoaderRegistryDebug.VERSION, acceptableRemoteVersions = "*")
 public class ModelLoaderRegistryDebug
 {
+    public static final boolean ENABLED = false;
     public static final String MODID = "forgedebugmodelloaderregistry";
     public static final String VERSION = "1.0";
+    private static Logger logger;
```
Comment (liach) [inline src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java:63]: In different test mods, the logger fields have their name in different cases. You need to unify them.

File: src/test/java/net/minecraftforge/fml/debug/FaultyCoreMod.java [hunk 5]
```diff
@@ -1,28 +1,47 @@
...
 import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;
 
-public class FaultyCoreMod implements IFMLLoadingPlugin {
+import java.util.Map;
+
+public class FaultyCoreMod implements IFMLLoadingPlugin
+{
     public static boolean enabled = false;
 
     public String[] getASMTransformerClass()
```
Comment (liach) [inline src/test/java/net/minecraftforge/fml/debug/FaultyCoreMod.java]: `@Override`?

File: src/test/java/net/minecraftforge/debug/BlockAiNodeTypeTest.java (hunk 1 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/debug/BlockAiNodeTypeTest.java:36]: This implementation of it never returns null

File: src/test/java/net/minecraftforge/debug/EnumPlantTypeTest.java (hunk 2 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/debug/EnumPlantTypeTest.java]: Auto-formatting. I'll remove this.

File: src/test/java/net/minecraftforge/debug/ItemTileDebug.java (hunk 3 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/debug/ItemTileDebug.java]: Missed it, thanks.

File: src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java (hunk 4 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java:63]: Ok

File: src/test/java/net/minecraftforge/fml/debug/FaultyCoreMod.java (hunk 5 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/fml/debug/FaultyCoreMod.java]: 👍

File: src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java (hunk 4 above)
Comment (mezz) [reply src/test/java/net/minecraftforge/debug/ModelLoaderRegistryDebug.java:63]: They are unified now.

**Review by liach — APPROVED** [review]

File: src/test/java/net/minecraftforge/debug/CanApplyAtEnchantingTableTest.java [hunk 6]
```diff
@@ -8,10 +8,10 @@
...
 import net.minecraftforge.fml.common.registry.GameRegistry;
 
-@Mod(modid="canapplyatenchantingtabletest", name="CanApplyAtEnchantingTableTest", version="0.0.0")
+@Mod(modid = "canapplyatenchantingtabletest", name = "CanApplyAtEnchantingTableTest", version = "0.0.0")
 public class CanApplyAtEnchantingTableTest
 {
-    public static final boolean ENABLE = true;
+    public static final boolean ENABLE = false;
 
     public static final Item testItem = new Item()
```
Comment (bs2609) [inline src/test/java/net/minecraftforge/debug/CanApplyAtEnchantingTableTest.java:16]: This needs `isItemTool` renaming to `isEnchantable`, and the missing `@Override` added.

Comment (mezz) [reply src/test/java/net/minecraftforge/debug/CanApplyAtEnchantingTableTest.java:16]: Fixed

**Comment by LexManos** [comment]
Do a quick pass to see if any clientSideOnly's are needed for connecting to a vanilla server, or vise versa.

**Comment by mezz** [comment]
Tested and any combination of vanilla + forge will work with the test mods now.
forge client + vanilla server
vanilla client + forge server
forge client + forge server

**Comment by mezz** [comment]
@LexManos this is good to merge now.

**Pull request merged**